lib/ansible/modules/cloud/cloudstack/cs_storage_ip_range.py

```diff
@@ -0,0 +1,305 @@
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: cs_storage_ip_range
+short_description: Manages Storage IP ranges on Apache CloudStack based clouds.
+description:
+    - Create updates and remove Storage IP ranges.
+version_added: "2.4"
+author: "Netservers Ltd. (@netservers)"
+options:
+  pod:
+    description:
+      - Pod name.
+    required: true
+  ipv4_start:
+    description:
+      - Start address for ipv4 range.
+    required: true
+    default: none
+  ipv4_end:
+    description:
+      - End address for ipv4 range.
+    required: false
+    default: none
+  ipv4_gateway:
+    description:
+      - Gateway IP for ipv4 range.
+    required: true
+    default: none
+  ipv4_netmask:
+    description:
+      - Gateway IP for ipv4 range.
+    required: true
+    default: none
+  vlan:
+    description:
+      - Optional. The vlan the ip range sits on, default to Null when it is not specificed
+      - This is mainly for Vmware as other hypervisors can directly retrieve bridge from physical network traffic type table
+    required: false
+    default: null
+  poll_async:
+    description:
+      - "Poll async jobs until job has finished."
+    required: false
+    default: true
+extends_documentation_fragment: cloudstack
+'''
+
+EXAMPLES = '''
+# Ensure a storage_ip_range is present
+- local_action:
+    module: cs_storage_ip_range
+    pod: pod01
+    ipv4_start: 1.0.0.10
+    ipv4_end: 1.0.0.100
+    ipv4_netmask: 255.255.255.0
+    ipv4_gateway: 1.0.0.1
+
+# Ensure a storage_ip_range is absent
+- local_action:
+    module: cs_storage_ip_range
+    pod: pod01
+    ipv4_start: 1.0.0.10
+    state: absent
+'''
+
+RETURN = '''
+---
+id:
+  description: UUID of the Storage IP range.
+  returned: success
+  type: string
+  sample: a3fca65a-7db1-4891-b97c-48806a978a96
+ipv4_start:
+  description: The start IP for the range.
+  returned: success
+  type: string
+  sample: 10.0.0.10
+ipv4_end:
+  description: The end IP for the range.
+  returned: success
+  type: string
+  sample: 10.0.0.20
+ipv4_gateway:
+  description: The gateway IP for the range.
+  returned: success
+  type: string
+  sample: 10.0.0.1
+ipv4_netmask:
+  description: The gateway IP for the range.
+  returned: success
+  type: string
+  sample: 255.255.255.0
+networkid:
+  description: The network UUID for the range.
+  returned: success
+  type: string
+  sample: a3fca65a-7db1-4891-b97c-48806a978a96
+podid:
```

Comment:
I don't see code returning podid, I usually do not return the ids for the arguments. When you looks at the integration tests, you'll see I compare the return values with the arguments to make sure the API used the same. I see no benefit in returning id for subelements.